src/Core/components/PiecewiseFunction.js

```diff
@@ -0,0 +1,793 @@
+import Function from './Function';
+import InlineComponent from './abstract/InlineComponent';
+import GraphicalComponent from './abstract/GraphicalComponent';
+import me from 'math-expressions';
+import { returnPiecewiseNumericalFunctionFromChildren } from '../utils/function';
+import { roundForDisplay } from '../utils/math';
+
+export default class PiecewiseFunction extends Function {
+  static componentType = "piecewiseFunction";
+
+
+  static createAttributesObject() {
+    let attributes = super.createAttributesObject();
+
+
+    delete attributes.nInputs;
+    delete attributes.nOutputs;
+    delete attributes.minima;
+    delete attributes.maxima;
+    delete attributes.extrema;
+    delete attributes.through;
+    delete attributes.throughSlope;
+
+
+
+    return attributes;
+  }
+
+  static returnSugarInstructions() {
+    return [];
+  };
+
+  static returnChildGroups() {
+
+    return [{
+      group: "functions",
+      componentTypes: ["function"]
+    }, {
+      group: "labels",
+      componentTypes: ["label"]
+    }]
+
+  }
+
+  static returnStateVariableDefinitions({ numerics }) {
+
+    let stateVariableDefinitions = super.returnStateVariableDefinitions({ numerics });
+
+    delete stateVariableDefinitions.isInterpolatedFunction;
+
+    stateVariableDefinitions.nInputs = {
+      public: true,
+      shadowingInstructions: {
+        createComponentOfType: "integer",
+      },
+      returnDependencies: () => ({}),
+      definition() {
+        return { setValue: { nInputs: 1 } }
+      }
+    }
+
+    stateVariableDefinitions.nOutputs = {
+      public: true,
+      shadowingInstructions: {
+        createComponentOfType: "integer",
+      },
+      returnDependencies: () => ({
+        functionChildren: {
+          dependencyType: "child",
+          childGroups: ["functions"],
+          variableNames: ["nOutputs"]
+        },
+      }),
+      definition({ dependencyValues }) {
+        if (dependencyValues.functionChildren.length > 0) {
+          return {
+            setValue: {
+              nOutputs: dependencyValues.functionChildren[0].stateValues.nOutputs
```

Comment:
Did you enforce somewhere that all of the children have the same number of outputs?

Reply:
No, I haven't explored what would happen if you gave it children with different numbers of inputs or even tested with more than one output.  I'd say the current version is designed just for functions with one output.  I would expect that bad things would happen if later functions had fewer outputs than the first and that extra output of later functions would be ignored.  At some point, we should come back and make piecewiseFunction be more robust.

Reply:
Added #2005 to address this.